moveit_ros/planning/planning_request_adapter_plugins/CMakeLists.txt

```
@@ -9,7 +11,7 @@ set(SOURCE_FILES
   src/resolve_constraint_frames.cpp
 )
 
-add_library(moveit_default_planning_request_adapter_plugins SHARED ${SOURCE_FILES})
+target_link_libraries(moveit_default_planning_request_adapter_plugins default_plan_request_adapter_parameters)
```

```suggestion
target_link_libraries(moveit_default_planning_request_adapter_plugins PRIVATE default_plan_request_adapter_parameters)
```

Should be PRIVATE I think, since it is not exposed in the interface. If it will let you, I know ament sometimes complains about this...

---

Yes :disappointed: I tried it and got 
```
  Starting >>> moveit_ros_planning
  --- stderr: moveit_ros_planning
  CMake Error at /opt/ros/rolling/share/ament_cmake_target_dependencies/cmake/ament_target_dependencies.cmake:145 (target_link_libraries):
    The keyword signature for target_link_libraries has already been used with
    the target "moveit_default_planning_request_adapter_plugins".  All uses of
    target_link_libraries with a target must be either all-keyword or
    all-plain.
  
    The uses of the keyword signature are here:
  
     * planning_request_adapter_plugins/CMakeLists.txt:14 (target_link_libraries)
  
  Call Stack (most recent call first):
    planning_request_adapter_plugins/CMakeLists.txt:17 (ament_target_dependencies)
  
  
  ---
  Failed   <<< moveit_ros_planning [8.90s, exited with code 1]
  Aborted  <<< moveit_planners_stomp [27.2s]
  
  Summary: 6 packages finished [1min 14s]
    1 package failed: moveit_ros_planning
    1 package aborted: moveit_planners_stomp
    3 packages had stderr output: moveit_common moveit_configs_utils moveit_ros_planning
    32 packages not processed
'build_target_workspace' returned with code '1' after 1 min 14 sec
```
I guess the `ament_target_dependencies` macro is the culprit since apparently, [you cannot call target_link_libs with PUBLIC and PRIVATE](https://stackoverflow.com/questions/47737558/uses-of-target-link-libraries-must-be-either-all-keyword-or-all-plain). Let's keep it public?